annotation/tests/test_jobs.py

```diff
@@ -343,4 +323,215 @@ def test_get_job_attributes_for_post_attribute_not_found():
+            files=[uuid.UUID("82533770-a99e-4873-8b23-6bbda86b59ae")],
+            validators=[],
+            owners=[uuid.UUID("82533770-a99e-4873-8b23-6bbda86b59ae")],
+            previous_jobs=[],
+            datasets=[1, 2],
+            is_auto_distribution=False,
+            categories=[],
+            deadline=None,
+            job_type=JobTypeEnumSchema.ExtractionJob,
+            tenant="test",
+        ),
+    )
+    assert result == expected_result
+
+
+def test_add_users():
+    mock_session = MagicMock()
+    users = [User(user_id=UUID(int=1)), User(user_id=UUID(int=2))]
+    new_user_ids = {UUID(int=3), UUID(int=4)}
+    expected_result = [
+        User(user_id=UUID(int=1)),
+        User(user_id=UUID(int=2)),
+        User(user_id=UUID(int=3)),
+    ]
+    mock_session.query().filter().all.return_value = [
+        User(user_id=UUID(int=3))
+    ]
+    result = annotation.jobs.services.add_users(
+        mock_session, users, new_user_ids
+    )
+    assert result == expected_result
+
+
+def test_update_job_categories(categories: Category):
+    mock_session = MagicMock()
+    patch_data = {"categories": [1, 2, 3]}
+    expected_result = ["18d3d189e73a4680bfa77ba3fe6ebee5"]
+    tenat = "test"
+    with patch(
+        "annotation.jobs.services.fetch_bunch_categories_db",
+        return_value=[categories],
+    ):
+        annotation.jobs.services.update_job_categories(
+            mock_session, patch_data, tenat
+        )
+        assert [cat.id for cat in patch_data["categories"]] == expected_result
+
+
+@pytest.mark.parametrize(
+    ("patch_data", "job"),
+    (
+        (
+            {
+                "extensive_coverage": 1,
+                "annotators": [
+                    User(user_id=UUID(int=1)),
+                    User(user_id=UUID(int=2)),
+                ],
+            },
+            Job(
+                annotators=[
+                    User(user_id=UUID(int=3)),
+                    User(user_id=UUID(int=4)),
+                ]
+            ),
+        ),
+        (
+            {"extensive_coverage": 1, "annotators": []},
+            Job(
+                annotators=[
+                    User(user_id=UUID(int=3)),
+                    User(user_id=UUID(int=4)),
+                ]
+            ),
+        ),
+        (
+            {
+                "extensive_coverage": 5,
+                "annotators": [
+                    User(user_id=UUID(int=1)),
+                    User(user_id=UUID(int=2)),
+                ],
+            },
+            Job(
+                annotators=[
+                    User(user_id=UUID(int=3)),
+                    User(user_id=UUID(int=4)),
+                ]
+            ),
+        ),
+        (
+            {"extensive_coverage": 5, "annotators": []},
+            Job(
+                annotators=[
+                    User(user_id=UUID(int=3)),
+                    User(user_id=UUID(int=4)),
+                ]
+            ),
+        ),
+    ),
+)
```

Comment:
Youre repeating users often. I think its best to have a fixture that returns all users for the tests and to put only user_id's in the parametrise. Then create the job inside of the test. Like: `job = Job(annotators=[User(user_id=id) for id in user_ids])`. I did this for a couple of my tests. Maybe it will make the whole test much cleaner.